src/utils/github-sync.ts

```diff
@@ -1,122 +1,142 @@
 import { Getter, useSetAtom } from "jotai"
 import { atomWithStorage, useAtomCallback } from "jotai/utils"
 import React from "react"
-import { githubRepoAtom, githubUserAtom, notesAtom, rawNotesAtom } from "../global-state"
-import { writeFiles } from "./github-fs"
+import { REPO_DIR, githubUserAtom, notesAtom, rawNotesAtom } from "../global-state"
+import git from "isomorphic-git"
+import http from "isomorphic-git/http/web"
+import { fs } from "./fs"
+import { logFileSystemState } from "./log-file-system"
 
 // Store SHA to avoid re-fetching notes if the SHA hasn't changed
 export const shaAtom = atomWithStorage<string | null>("sha", null)
 
 const githubUserCallback = (get: Getter) => get(githubUserAtom)
-const githubRepoCallback = (get: Getter) => get(githubRepoAtom)
-
 const notesCallback = (get: Getter) => get(notesAtom)
 
 export function useRenameTag() {
   const getGitHubUser = useAtomCallback(githubUserCallback)
-  const getGitHubRepo = useAtomCallback(githubRepoCallback)
   const getNotes = useAtomCallback(notesCallback)
   const setRawNotes = useSetAtom(rawNotesAtom)
 
   return React.useCallback(
     async (oldName: string, newName: string) => {
       const notes = getNotes()
 
-      const updatedRawNotes = [...notes.values()]
-        // Notes that contain the old tag
-        .filter((note) => note.tags.includes(oldName))
-        .reduce<Record<string, string>>((updatedRawNotes, note) => {
-          // Find and replace the old tag with the new tag
-          updatedRawNotes[note.id] = note.content.replace(`#${oldName}`, `#${newName}`)
-          return updatedRawNotes
-        }, {})
+      const updatedNotesContent: Record<string, string> = {}
+      for (const note of notes.values()) {
+        if (note.tags.includes(oldName)) {
+          const newContent = note.content.replace(`#${oldName}`, `#${newName}`)
```

Comment:
This only renames inline (`#mytag`). We should also support renaming tags defined in the frontmatter of notes:

```
---
tags: [mytag]
---
```

This was a bug with the existing implementation but it would be nice to address it now.